modules/phase_field/src/kernels/ACBarrierFunction.C

```diff
@@ -74,12 +76,16 @@ ACBarrierFunction::computeQpOffDiagJacobian(unsigned int jvar)
     if (i != j)
       sum_etai2 += (*_vals[i])[_qp] * (*_vals[i])[_qp];
 
-  df0deta_base = (*_vals[j])[_qp] * (*_vals[j])[_qp] - 1.0 +
-                 2.0 * _gamma[_qp] * (_u[_qp] * _u[_qp] + sum_etai2);
-  df0deta = (*_vals[j])[_qp] * df0deta_base;
-
-  return ((*_d2mudvardeta[j])[_qp] * calculateF0() + _dmudvar[_qp] * df0deta) * _phi[_j][_qp] *
-         _test[_i][_qp] * _L[_qp];
+  // prevent index from exceeding # of variables
+  if (j < _vals.size())
```

Comment:
It seems to me the correct fix would be to use `getParameterJvarMap` from JVarMapKernelInterface instead.